pig/src/main/java/org/apache/zeppelin/pig/PigUtils.java

```diff
@@ -0,0 +1,292 @@
+    LOGGER.debug("Create pig script file:" + tmpFile.getAbsolutePath());
+    FileWriter writer = new FileWriter(tmpFile);
+    IOUtils.write(content, writer);
+    writer.close();
+    return tmpFile.getAbsoluteFile();
+  }
+
+  public static File createTempPigScript(List<String> lines) throws IOException {
+    return createTempPigScript(StringUtils.join(lines, "\n"));
+  }
+
+  public static String extactJobStats(PigStats stats) {
+    if (stats instanceof SimplePigStats) {
+      return extractFromSimplePigStats((SimplePigStats) stats);
+    } else if (stats instanceof TezPigScriptStats) {
+      return extractFromTezPigStats((TezPigScriptStats) stats);
+    } else {
+      throw new RuntimeException("Unrecognized stats type:" + stats.getClass().getSimpleName());
+    }
+  }
+
+  public static String extractFromSimplePigStats(SimplePigStats stats) {
+
+    try {
+      Field userIdField = PigStats.class.getDeclaredField("userId");
+      userIdField.setAccessible(true);
+      String userId = (String) (userIdField.get(stats));
+      Field startTimeField = PigStats.class.getDeclaredField("startTime");
+      startTimeField.setAccessible(true);
+      long startTime = (Long) (startTimeField.get(stats));
+      Field endTimeField = PigStats.class.getDeclaredField("endTime");
+      endTimeField.setAccessible(true);
+      long endTime = (Long) (endTimeField.get(stats));
+
+      if (stats.getReturnCode() == PigRunner.ReturnCode.UNKNOWN) {
+        LOGGER.warn("unknown return code, can't display the results");
+        return null;
+      }
+      if (stats.getPigContext() == null) {
+        LOGGER.warn("unknown exec type, don't display the results");
+        return null;
+      }
+
+      SimpleDateFormat sdf = new SimpleDateFormat(DATE_FORMAT);
+      StringBuilder sb = new StringBuilder();
+      sb.append("\nHadoopVersion\tPigVersion\tUserId\tStartedAt\tFinishedAt\tFeatures\n");
+      sb.append(stats.getHadoopVersion()).append("\t").append(stats.getPigVersion()).append("\t")
+              .append(userId).append("\t")
+              .append(sdf.format(new Date(startTime))).append("\t")
+              .append(sdf.format(new Date(endTime))).append("\t")
+              .append(stats.getFeatures()).append("\n");
+      sb.append("\n");
+      if (stats.getReturnCode() == PigRunner.ReturnCode.SUCCESS) {
+        sb.append("Success!\n");
+      } else if (stats.getReturnCode() == PigRunner.ReturnCode.PARTIAL_FAILURE) {
+        sb.append("Some jobs have failed! Stop running all dependent jobs\n");
+      } else {
+        sb.append("Failed!\n");
+      }
+      sb.append("\n");
+
+      Field jobPlanField = PigStats.class.getDeclaredField("jobPlan");
+      jobPlanField.setAccessible(true);
+      PigStats.JobGraph jobPlan = (PigStats.JobGraph) jobPlanField.get(stats);
+
+      if (stats.getReturnCode() == PigRunner.ReturnCode.SUCCESS
+              || stats.getReturnCode() == PigRunner.ReturnCode.PARTIAL_FAILURE) {
```

Comment:
just that `stats.getReturnCode() == PigRunner.ReturnCode.PARTIAL_FAILURE` is in both the "SUCCESS" and "FAILURE" cases. So if I understand correctly, if return code is PARTIAL_FAILURE, it will say "Job Stats" and "Failed Jobs" at the same time